certora/specs/ERC20.spec

```diff
@@ -21,6 +21,12 @@ ghost mathint sumOfBalances {
     init_state axiom sumOfBalances == 0;
 }
 
+// Because `balance` has a uint256 type, any balance addition in CVL1 behaved as a `require_uint256()` casting,
+// leaving out the possibility of overflow. This is not the case in CVL2 where casting became more explicit.
+// A counterexample in CVL2 is having an initial state where Alice initial balance is larger than totalSupply, which 
```

Comment:
Counterexample to what ? to a rule ? an invariant ?

We should have an invariant that we can require and that would say that Alice's balance cannot be bigger than the totalSupply.

Said otherwise, that is a property we want to prove, not assume.

Reply:
> Counterexample to what ? to a rule ? an invariant ?

To any rule/invariant checking for balance consistency.

I mostly agree with you. I recall trying to define this `require` as an invariant but it was again failing because the invariant can't be proved for a run similar to the one I described in [this convo](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4527#discussion_r1320028525).

Defining an invariant such that `balance[<some address>] <= sumOfBalances` may work for only one address but you'd need to apply it to every address with a `forall` quantifier, but I couldn't manage to make it work.